.github/workflows/supported_storages_compatibility_check.yaml

```diff
@@ -0,0 +1,728 @@
+    runs-on: ubuntu-latest
+
+    services:
+      sqlserver:
+        image: mcr.microsoft.com/mssql/server:2019-latest
+        env:
+          MSSQL_PID: "Express"
+          SA_PASSWORD: "SqlServer19"
+          ACCEPT_EULA: "Y"
+        ports:
+          - 1433:1433
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Setup and execute Gradle 'integrationTestJdbc' task
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: integrationTestJdbc -Dscalardb.jdbc.url=jdbc:sqlserver://localhost:1433;encrypt=true;trustServerCertificate=true -Dscalardb.jdbc.username=sa -Dscalardb.jdbc.password=SqlServer19
+
+      - name: Upload Gradle test reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: jdbc_sqlserver_2019_integration_test_reports
+          path: core/build/reports/tests/integrationTestJdbc
+
+  jdbc-sqlserver-2022-integration-test:
+    runs-on: ubuntu-latest
+
+    services:
+      sqlserver:
+        image: mcr.microsoft.com/mssql/server:2022-latest
+        env:
+          MSSQL_PID: "Express"
+          SA_PASSWORD: "SqlServer22"
+          ACCEPT_EULA: "Y"
+        ports:
+          - 1433:1433
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Setup and execute Gradle 'integrationTestJdbc' task
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: integrationTestJdbc -Dscalardb.jdbc.url=jdbc:sqlserver://localhost:1433;encrypt=true;trustServerCertificate=true -Dscalardb.jdbc.username=sa -Dscalardb.jdbc.password=SqlServer22
+
+      - name: Upload Gradle test reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: jdbc_sqlserver_2022_integration_test_reports
+          path: core/build/reports/tests/integrationTestJdbc
+
+  jdbc-sqlite-integration-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Set up SQLite3
+        run: sudo apt-get install -y sqlite3
+
+      - name: Setup and execute Gradle 'integrationTestJdbc' task
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: integrationTestJdbc -Dscalardb.jdbc.url=jdbc:sqlite:integration.sqlite3?busy_timeout=50000
+
+      - name: Upload Gradle test reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: jdbc_sqlite_integration_test_reports
+          path: core/build/reports/tests/integrationTestJdbc
+
+  jdbc-mariadb-integration-test:
```

Comment:
@feeblefakie If we officially support MariaDB, we need to decide support versions. What do you think?

Reply:
@feeblefakie (cc @brfrn169)
After discussing with @brfrn169, MariaDB only supports the major versions 10 and 11 presently, so we are considering supporting them both.
cf. the list of MariaDB version and their EOL https://endoflife.date/mariadb

I will revise the workflow to test with the latest `10.x` and `11.x` versions, but please let us know if you have any concerns.

Reply:
I tested MariaDB 11 and MariaDB 10, and ScalarDB is not compatible with all versions :
- 10.0 to 10.3: ❌ it fails  
- 10.4 to 10.11: ✅ it works  
- 11.0: ❌ it fails  
- 11.1-rc: ✅ which is currently only available as a release candidate version, it is working. I suppose a production version will be released soon. 

For these reasons, we decided to only mention in the supported storage for now MariaDB 10 but not MariaDB 11. 
Once MariaDB 11.1 is released officially, I think we can add it to the list of supported versions.